src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -0,0 +1,1546 @@
+     * so).
+     *
+     * <p> Ideally, all {@link LeapableGenerator} objects produced by iterative
+     * leaping and jumping from a single original {@link LeapableGenerator}
+     * object are statistically independent of one another and individually
+     * uniform. In practice, one must settle for some approximation to
+     * independence and uniformity. In particular, a specific implementation may
+     * assume that each generator in a stream produced by the {@code leaps}
+     * method is used to produce (by jumping) a number of objects no larger than
+     * 2<sup>64</sup>. Implementors are advised to use algorithms whose period
+     * is at least 2<sup>191</sup>.
+     *
+     * <p> Methods are provided to perform a single leap operation and also to
+     * produce a stream of generators produced from the original by iterative
+     * copying and leaping of internal state. The generators produced must
+     * implement the {@link JumpableGenerator} interface but need not also
+     * implement the {@link LeapableGenerator} interface. A typical strategy for
+     * a multithreaded application is to create a single
+     * {@link LeapableGenerator} object, calls its {@code leaps} method exactly
+     * once, and then parcel out generators from the resulting stream, one to
+     * each thread. Then the {@link JumpableGenerator#jump() jump}() method of
+     * each such generator be called to produce a substream of generator
+     * objects.
+     *
+     * <p> Objects that implement {@link LeapableGenerator} are typically not
+     * cryptographically secure. Consider instead using {@link SecureRandom} to
+     * get a cryptographically secure pseudo-random number generator for use by
+     * security-sensitive applications.
+     */
+    interface LeapableGenerator extends JumpableGenerator {
+
+        /**
+         * Returns an instance of {@link LeapableGenerator} that utilizes the
+         * {@code name} <a href="package-summary.html#algorithms">algorithm</a>.
+         *
+         * @param name  Name of random number generator
+         *              <a href="package-summary.html#algorithms">algorithm</a>
+         *
+         * @return An instance of {@link LeapableGenerator}
+         *
+         * @throws NullPointerException if name is null
+         * @throws IllegalArgumentException if the named algorithm is not found
+         */
+        static LeapableGenerator of(String name) {
+            Objects.requireNonNull(name);
+
+            return RandomGeneratorFactory.of(name, LeapableGenerator.class);
+        }
+
+        /**
+         * Returns a new generator whose internal state is an exact copy of this
+         * generator (therefore their future behavior should be identical if
+         * subjected to the same series of operations).
+         *
+         * @return a new object that is a copy of this generator
+         */
+        LeapableGenerator copy();
```

Comment:
Suggest adding an Override annotation here and possibly to inheritDoc the text from Jumpable.copy.

Reply:
Fails due to result variance.